src/paperoni/config.py

```diff
@@ -28,10 +28,40 @@ class PaperoniConfig:
     api_keys: Keys[str, str] = field(default_factory=Keys)
     fetch: TaggedSubclass[Fetcher] = field(default_factory=RequestsFetcher)
     focuses: Focuses = field(default_factory=Focuses)
+    autofocus: AutoFocus[str, AutoFocus.Type] = field(default_factory=AutoFocus)
     refine: Refine = None
     work_file: Path = None
     collection: TaggedSubclass[PaperCollection] = None
 
+    def __post_init__(self):
+        # Only used for type hinting
+        self._file: Path = getattr(self, "_file", None)
+        # Ideally we would not need this but there seams to an issue with
+        # autofocus not being of type AutoFocus when using something like:
+        # with gifnoc.overlay(
+        #         {
+        #             "paperoni.autofocus": {
+        #                 "author": {"score": 1, "threshold": 5},
+        #                 "institution": {"score": 5, "threshold": 20},
+        #             },
+        #         }
+        #     ):
+        self.autofocus = AutoFocus(**self.autofocus)
+
+    # TODO: Why does this seams to disable future gifnoc.define like
+    # `paperoni.semantic_scholar`?
+    # gifnoc.proxy.MissingConfigurationError: No configuration was found for 'paperoni.semantic_scholar
```

Comment:
gifnoc.define modifies the data structure in order to stash additional fields and it is designed to work even *after* the configuration is initially deserialized. I'm not surprised it's fragile with respect to serieux_deserialize.